### PR TITLE
Set active tab to history when navigating from transfer push

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #1190

Changes
- Ellipsize long token names for amounts
- Use en layout for currency names
- Open history tx tab when navigation from push notifications about transfers

@gnosis/mobile-devs

### CODE REVIEW

File: app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt [hunk 1]
```diff
@@ -36,7 +36,7 @@ class FiatListAdapter : RecyclerView.Adapter<FiatListAdapter.FiatViewHolder>() {
 
     override fun onBindViewHolder(holder: FiatViewHolder, position: Int) {
         val item = items[position]
-        val displayName = Currency.getInstance(item).displayName
+        val displayName = Currency.getInstance(item).getDisplayName(Locale("en", Locale.getDefault().country))
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt:39]: Ah cool you fixed the issue with the fiat names #1190  ?

File: app/src/main/res/layout/item_tx_transfer.xml [hunk 2]
```diff
@@ -47,6 +47,8 @@
         android:layout_marginStart="@dimen/default_small_margin"
         android:layout_marginEnd="@dimen/default_small_margin"
         android:gravity="end"
+        android:ellipsize="end"
+        android:maxLines="1"
```
Comment (jpalvarezl) [inline app/src/main/res/layout/item_tx_transfer.xml:51]: Would we need to give the same treatment to Custom transactions? As the method name could also be "unreasonably" long

File: app/src/main/java/io/gnosis/safe/ui/StartActivity.kt [hunk 3]
```diff
@@ -66,9 +66,10 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
                         safeRepository.setActiveSafe(it)
                         setSafeData(it)
                     }
-                    //TODO: figure out which tab (QUEUE vs HISTORY) to open
                     if (txId == null) {
-                        Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.transactionsFragment)
+                        Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.transactionsFragment, Bundle().apply {
+                            putInt("activeTab", 1) // open history tab
```
Comment (jpalvarezl) [inline app/src/main/java/io/gnosis/safe/ui/StartActivity.kt:71]: Would be cool to use an `enum` for this.

**Review by jpalvarezl — APPROVED** [review]

File: app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt (hunk 1 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt:39]: yes

File: app/src/main/res/layout/item_tx_transfer.xml (hunk 2 above)
Comment (elgatovital) [reply app/src/main/res/layout/item_tx_transfer.xml:51]: we do it already for custom tx

File: app/src/main/java/io/gnosis/safe/ui/StartActivity.kt [hunk 4]
```diff
@@ -66,9 +67,10 @@ class StartActivity : BaseActivity(), SafeOverviewNavigationHandler {
                         safeRepository.setActiveSafe(it)
                         setSafeData(it)
                     }
-                    //TODO: figure out which tab (QUEUE vs HISTORY) to open
                     if (txId == null) {
-                        Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.transactionsFragment)
+                        Navigation.findNavController(this@StartActivity, R.id.nav_host).navigate(R.id.transactionsFragment, Bundle().apply {
+                            putInt("activeTab", TxPagerAdapter.Tabs.HISTORY.ordinal) // open history tab
+                        })
```
Comment (biafra23) [inline app/src/main/java/io/gnosis/safe/ui/StartActivity.kt:73]: For assets and settings we do not want this ATM?

File: app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt (hunk 1 above)
Comment (jpalvarezl) [reply app/src/main/java/io/gnosis/safe/ui/settings/app/fiat/FiatListAdapter.kt:39]: thanks :D

File: app/src/main/java/io/gnosis/safe/ui/StartActivity.kt (hunk 4 above)
Comment (elgatovital) [reply app/src/main/java/io/gnosis/safe/ui/StartActivity.kt:73]: what do you mean? We have those enums also for assets and settings

**Review by biafra23 — APPROVED** [review]

**Pull request merged**